### DISCUSSION TITLE
How to build for production and optimize package size?

### DISCUSSION THREAD

**Discussion by oedev-apps** [quasarframework/quasar#16662] [General - Components / Directives / etc]
Using Vue3 Quasar CLI with Vite - 

What is the preferred / suggested way to ensure dev only components, pages, etc. are not included in production builds?

For example, let's say that I'm building a capacitor app. In dev, I have some extra components that allow me to inspect the state, access dev only features, etc. So when I run "quasar run dev" I will see these components. When I build the app for production - "quasar build -m android", I don't want the code for these components included in the package.

**Comment by metalsadman** [comment]
@oedev-apps start here and the rest of that docs https://quasar.dev/quasar-cli-vite/handling-process-env#example
